### DISCUSSION TITLE
Python decouple

### DISCUSSION THREAD

**Discussion by yeboahd24** [Infisical/infisical#146] [Q&A] [answered]
Hi team
I am having issue of the .`env` file actually I am trying to use it with my django project I have followed the procedure and is okay but my `.env` file is not working.
I am using python decouple I don't know if is the reason why my environment file is not working.
I am always getting missing variable meanwhile everything alright, when I run it without infisical it works alright.

**Comment by vmatsiiako** [comment]
@yeboahd24 Thank you for submitting this discussion! I think @maidul98 will try to look into this, and get back to you shortly :)

**Reply by yeboahd24** [reply]
Okay thanks for your response.

**Reply by maidul98** [reply]
Yes, I will work on reproducing this this issue. Can you share a code snippet of how you are accessing the environment variable in code?

**Reply by yeboahd24** [reply]
Here is it. `STRIPE_PUBLIC_KEY = config("STRIPE_PUBLIC_KEY")
`

**Reply by vmatsiiako** [reply]
HI @yeboahd24! We're a little overwhelmed with different requests, so it might take a bit longer to debug your issue. We will update this discussion when we find a solution. 

Meanwhile, if you find a solution, could you please let us know here. Thank you :)

**Comment by maidul98** [comment]
I was able to run my django app with `infisical run -- python manage.py runserver` and use python decouple with any issues. Can you let me know if you see your secrets when you print `print(os.environ.items())`? Meanwhile, I will also verify on a Ubuntu as I know that is what you are using

**Comment by yeboahd24** [comment]
`ItemsView(environ({'SHELL': '/bin/bash', 'SESSION_MANAGER': 'local/hp-notebook:@/tmp/.ICE-unix/3603,unix/hp-notebook:/tmp/.ICE-unix/3603', 'QT_ACCESSIBILITY': '1', 'COLORTERM': 'truecolor', 'XDG_CONFIG_DIRS': '/etc/xdg/xdg-ubuntu:/etc/xdg', 'SSH_AGENT_LAUNCHER': 'gnome-keyring', 'XDG_MENU_PREFIX': 'gnome-', 'GNOME_DESKTOP_SESSION_ID': 'this-is-deprecated', 'GNOME_SHELL_SESSION_MODE': 'ubuntu', 'SSH_AUTH_SOCK': '/run/user/1000/keyring/ssh', 'XMODIFIERS': '@im=ibus', 'DESKTOP_SESSION': 'ubuntu', 'GTK_MODULES': 'gail:atk-bridge', 'PWD': '/home/dominic/MVP/mvp', 'LOGNAME': 'dominic', 'XDG_SESSION_DESKTOP': 'ubuntu', 'XDG_SESSION_TYPE': 'x11', 'GPG_AGENT_INFO': '/run/user/1000/gnupg/S.gpg-agent:0:1', 'SYSTEMD_EXEC_PID': '3624', 'XAUTHORITY': '/run/user/1000/gdm/Xauthority', 'WINDOWPATH': '2', 'HOME': '/home/dominic', 'USERNAME': 'dominic', 'LANG': 'en_US.UTF-8', 'LS_COLORS': 'rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.webp=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:', 'XDG_CURRENT_DESKTOP': 'ubuntu:GNOME', 'VIRTUAL_ENV': '/home/dominic/mvp', 'VTE_VERSION': '6800', 'GNOME_TERMINAL_SCREEN': '/org/gnome/Terminal/screen/38db98c1_7a80_4ecd_bf83_f369374262ef', 'DJANGO_SETTINGS_MODULE': 'mvp_project.settings.local', 'LESSCLOSE': '/usr/bin/lesspipe %s %s', 'XDG_SESSION_CLASS': 'user', 'TERM': 'xterm-256color', 'LESSOPEN': '| /usr/bin/lesspipe %s', 'USER': 'dominic', 'GNOME_TERMINAL_SERVICE': ':1.5388', 'DISPLAY': ':1', 'SHLVL': '1', 'QT_IM_MODULE': 'ibus', 'VIRTUAL_ENV_PROMPT': '(mvp) ', 'XDG_RUNTIME_DIR': '/run/user/1000', 'PS1': '(mvp) \\[\\e]0;\\u@\\h: \\w\\a\\]${debian_chroot:+($debian_chroot)}\\[\\033[01;32m\\]\\u@\\h\\[\\033[00m\\]:\\[\\033[01;34m\\]\\w\\[\\033[00m\\]\\$ ', 'XDG_DATA_DIRS': '/usr/share/ubuntu:/usr/share/gnome:/usr/local/share/:/usr/share/:/var/lib/snapd/desktop', 'PATH': '/home/dominic/mvp/bin:/home/dominic/.cargo/bin:/home/dominic/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin', 'GDMSESSION': 'ubuntu', 'DBUS_SESSION_BUS_ADDRESS': 'unix:path=/run/user/1000/bus', 'OLDPWD': '/home/dominic', '_': '/home/dominic/mvp/bin/python'}))
`

Here is I am getting which is not my .env file

**Reply by maidul98** [reply]
And to confirm you are using `infisical run -- python manage.py runserver` command? Also when you enter that command do you see something like this?
![Screen Shot 2022-12-22 at 7 25 29 PM](https://user-images.githubusercontent.com/9300960/209247140-6f601ff0-c8ef-47e8-8acf-3cf65e256694.png)

**Reply by yeboahd24** [reply]
Yes, I see that.

**Comment by maidul98** [comment, marked as answer]
I was also able to get it to work on my Ubuntu machine. If you would like, we do a debug session together over zoom. Let me know if you like that

**Reply by yeboahd24** [reply]
Okay what about google meet

**Reply by maidul98** [reply]
That works. What time would you like to do it? I am available for the next 30 minutes. Or feel free to send me an invite to my email maidul@infisical.com for EST time

**Reply by yeboahd24** [reply]
Send me invitation now and let connect.

**Reply by maidul98** [reply]
https://meet.google.com/sfs-kwoq-mcb

**Reply by maidul98** [reply]
Happy to have solved your issue! Let us know if you face any other trouble. 

For future readers, the issue was missing ENVS in Infisical project

**Reply by yeboahd24** [reply]
Thanks for your time bro.
